src/ops/dbcsr_io.F

```diff
@@ -580,34 +580,63 @@ SUBROUTINE printmat_s(matrix, rows, cols, iunit, title, tr)
    END SUBROUTINE printmat_s
 
 ! **************************************************************************************************
-!> \brief ...
+!> \brief Prints the elements of a matrix.
 !> \param matrix ...
-!> \param rows ...
-!> \param cols ...
+!> \param rows the logical (possibly detransposed) matrix
+!>                            size, not the stored size
+!> \param cols the logical (possibly detransposed) matrix
+!>                            size, not the stored size
 !> \param iunit ...
 !> \param title ...
-!> \param tr ...
+!> \param[in] tr    (optional) specifies whether the elements are stored
+!>                  transposed
 ! **************************************************************************************************
-   SUBROUTINE printmat_d(matrix, rows, cols, iunit, title, tr)
-      REAL(KIND=real_8), DIMENSION(:), INTENT(IN)        :: matrix
+   SUBROUTINE printmat_c(matrix, rows, cols, iunit, title, tr)
+      complex(KIND=real_4), DIMENSION(:), INTENT(IN)        :: matrix
       INTEGER, INTENT(IN)                                :: rows, cols, iunit
       CHARACTER(*), INTENT(IN), OPTIONAL                 :: title
       LOGICAL, INTENT(IN), OPTIONAL                      :: tr
 
-      IF (PRESENT(title)) THEN
-         IF (PRESENT(tr)) THEN
-            CALL printmat_s(REAL(matrix, KIND=sp), rows, cols, iunit, title, tr)
-         ELSE
-            CALL printmat_s(REAL(matrix, KIND=sp), rows, cols, iunit, title)
+      CHARACTER(30)                                      :: f
+      INTEGER                                            :: m, n, r, c
+      LOGICAL                                            :: t
+      complex(KIND=dp)                                      :: bit_bucket
+
+!   ---------------------------------------------------------------------------
+
+      m = rows
+      n = cols
+      t = .FALSE.
+      IF (PRESENT(title)) WRITE (iunit, *) title
+      IF (PRESENT(tr)) THEN
+         IF (tr) THEN
+            t = .TRUE.
+            m = cols
+            n = rows
          ENDIF
-      ELSE
-         IF (PRESENT(tr)) THEN
-            CALL printmat_s(REAL(matrix, KIND=sp), rows, cols, iunit, tr=tr)
+      ENDIF
+      DO r = LBOUND(matrix, 1), UBOUND(matrix, 1)
+         bit_bucket = matrix(r)
+      ENDDO
+      bit_bucket = 0.0_dp
+      DO r = LBOUND(matrix, 1), UBOUND(matrix, 1)
+         bit_bucket = bit_bucket + matrix(r)
+      ENDDO
+      IF (m .GT. 10000) m = 0
```

Comment:
?

Reply:
what is wrong here?

Reply:
magic constants

Reply:
I copied it from the above `print_matrix_s` , it is the same there.